sdks/wasm/packager.cs

```diff
@@ -489,7 +498,7 @@ class WasmOptions {
 		}
 
 		if (enable_aot) {
-			var to_aot = new Dictionary<string, bool> ();
+			var to_aot = new Dictionary<string, bool> (StringComparer.OrdinalIgnoreCase);
```

Comment:
is it possible to load two assemblies named Foo.dll and FOO.dll? Is it reasonable to expect someone would do that? I'm okay with the answer being "don't do that" if they try to feed them to the linker, just wondering though

Reply:
That's a good question. The main reason I'm adding this is that assembly names are provided by user entry in my scenario, but I'm not sure how that's relevant. 

I should probably do that validation/sanitization upfront.

Reply:
.NET Framework (and Mono, and Core (?)) treat assembly names as case insensitive (so if an assembly names FOO is already loaded, references to Foo will just give you back FOO again).  But Mono probes the filesystem in a case-preserving manner (so if Foo and FOO aren't loaded and you ask for FOO.dll, we'll look for that exact name and won't see Foo.dll).

You could, sometimes, maybe, load two assemblies with the same name if you use `Assembly.LoadFile` or `Assembly.Load (byte[])` - the guarantees about what will happen are pretty vague, however.

Reply:
I remember now. The reason I added this was for convenience, as the user needs to input assemblies to AOT manually (even if those are discovered through assemblies inspection) through the CLI. That's the same reason I added this (to skip AOT): 

https://github.com/mono/mono/pull/13334/files/828793614595bab0038181daeaf6494780230095#diff-6ada20ee030b151fa1dba678cfaf43e4R522

as it can be difficult to the user to know the exact casing of an assembly.